src/services/types.ts

```diff
@@ -71,6 +71,10 @@ namespace ts {
         getLineAndCharacterOfPosition(pos: number): LineAndCharacter;
     }
 
+    export interface SourceMapSource {
+        /* @internal */ getLineAndCharacterOfPosition(pos: number): LineAndCharacter;

```

Comment:
`source` is optional on `SourceMapRange` and a `SourceMapSource` should always be created by calling `createSourceMapSource` in _factory.ts_. Marking it internal here means that `SourceMapSource` can't be used as a `SourceFileLike` in the public API, only in the compiler.